modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/GossipDaemon.scala

```diff
@@ -95,6 +99,23 @@ object GossipDaemon {
           }
       }
 
+    private def validateCollateral(batch: RumorBatch): F[RumorBatch] =
+      batch.filterA {
+        case (hash, signedRumor) => {
+          signedRumor.proofs.toNonEmptyList
+            .map(proof => collateral.hasCollateral(PeerId.fromId(proof.id)))
+            .sequence
+            .map(_.foldLeft(true)(_ && _))
+            .flatTap { valid =>
+              if (valid) Applicative[F].unit
```

Comment:
You can use whenA on application here

Reply:
Changed, but not to whenA but to unlessA. Also changed in previous validation methods.